File: Catalog/Catalog.Host/Models/Dtos/Artefact.cs
namespace Catalog.Host.Models.Dtos
{
    public class Artefact
    {
        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public decimal Cost { get; set; }
        public string Nature { get; set; } = null!;
        public string ImageUrl { get; set; } = null!;

        public Characteristic Values { get; set; } = null!;
        public Anomaly Anomaly { get; set; } = null!;
        public AbnormalType Type { get; set; } = null!;
        public Frequency Meets { get; set; } = null!;
    }
}

File: Web/webapp/src/interfaces/artefact.ts
import { IAnomaly } from "./anomaly";
import { ICharacteristic } from "./characteristic";
import { IMeet } from "./meets";
import { IType } from "./type";

export interface IArtefact {
    'id': number,
    'name': string,
    'cost': number,
    'nature': string,
    'imageUrl': string,
    'values': ICharacteristic,
    'anomaly': IAnomaly
    'type': IType,
    'meets': IMeet,
};